src/DMXSerial2.cpp

```diff
@@ -805,6 +814,78 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength) {
+            // Out of range sensor
+            nackReason = E120_NR_DATA_OUT_OF_RANGE;
+          } else {
+            _rdm.packet.DataLength = 13 + strlen(_initData->sensors[sensorNr].description);
+            _rdm.packet.Data[0] = sensorNr;
+            _rdm.packet.Data[1] = _initData->sensors[sensorNr].type;
+            _rdm.packet.Data[2] = _initData->sensors[sensorNr].unit;
+            _rdm.packet.Data[3] = _initData->sensors[sensorNr].prefix;
+            WRITEINT(_rdm.packet.Data +  4, _initData->sensors[sensorNr].rangeMin);
+            WRITEINT(_rdm.packet.Data +  6, _initData->sensors[sensorNr].rangeMax);
+            WRITEINT(_rdm.packet.Data +  8, _initData->sensors[sensorNr].normalMin);
+            WRITEINT(_rdm.packet.Data + 10, _initData->sensors[sensorNr].normalMax);
+            _rdm.packet.Data[12] = (_initData->sensors[sensorNr].lowHighSupported ? 2 : 0) | (_initData->sensors[sensorNr].recordedSupported ? 1 : 0);
+            memcpy(_rdm.packet.Data + 13, _initData->sensors[sensorNr].description, _rdm.packet.DataLength - 13);
+            handled = true;
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
+      }
+    } else if (Parameter == SWAPINT(E120_SENSOR_VALUE)) { // 0x0201
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength != 1) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength) {
+            // Out of range sensor
+            nackReason = E120_NR_DATA_OUT_OF_RANGE;
+          } else {
+            int16_t sensorValue = 0;
+            int16_t lowestValue = 0;
+            int16_t highestValue = 0;
+            int16_t recordedValue = 0;
+            bool8 res = false;
+            if (_sensorFunc) {
+              res = _sensorFunc(sensorNr, &sensorValue, &lowestValue, &highestValue, &recordedValue);
+            }
+            if (res) {
+              _rdm.packet.DataLength = 9;
+              _rdm.packet.Data[0] = sensorNr;
+              WRITEINT(_rdm.packet.Data +  1, sensorValue);
+              WRITEINT(_rdm.packet.Data +  3, lowestValue);
+              WRITEINT(_rdm.packet.Data +  5, highestValue);
+              WRITEINT(_rdm.packet.Data +  7, recordedValue);
+              handled = true;
+            } else {
+              nackReason = E120_NR_HARDWARE_FAULT;
+            }
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
```

Comment:
Not unexpected, SET lets you clear/reset the recorded info. Here 0xff (all) clears for all sensors.